### PR TITLE
Vector search feature support flag

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Vector search enabled flag in bridge.
Vector search flag added to document properties in api commons.

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/DataStoreConfig.java [hunk 1]
```diff
@@ -51,4 +51,7 @@ public interface DataStoreConfig {
   /** @return If logged batches are enabled, defaults to <code>true</code>. */
   @WithDefault("true")
   boolean loggedBatchesEnabled();
+
+  @WithDefault("true")
+  boolean vectorSearchEnabled();
```
Comment (kathirsvn) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/config/DataStoreConfig.java:56]: Missing java doc

File: coordinator/persistence-api/src/main/java/io/stargate/db/Persistence.java [hunk 2]
```diff
@@ -96,6 +96,10 @@ default boolean supportsSecondaryIndex() {
   /** Returns true if the persistence backend supports Storage Attached Indexes. */
   boolean supportsSAI();
 
+  default boolean supportsVectorSearch() {
```
Comment (kathirsvn) [inline coordinator/persistence-api/src/main/java/io/stargate/db/Persistence.java:99]: Missing java doc

**Review by kathirsvn — APPROVED** [review]
LGTM except a couple of javadoc updates required

**Review by jeffreyscarpenter — APPROVED** [review]

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

**Pull request merged**